src/main/java/uk/gov/pay/api/service/AgreementService.java

```diff
@@ -69,33 +72,45 @@ private AgreementLinks createLinksFromMandateResponse(MandateConnectorResponse m
         agreementLinks.addKnownLinksValueOf(mandate.getLinks());
         return agreementLinks;
     }
-    
-    private boolean isFound(Response connectorResponse) {
-        return connectorResponse.getStatus() == HttpStatus.SC_OK;
-    }
 
-    private boolean isCreated(Response connectorResponse) {
-        return connectorResponse.getStatus() == HttpStatus.SC_CREATED;
-    }
-
-    private Response createAgreement(Account account, CreateAgreementRequest createAgreementRequest) {
+    Response createMandate(Account account, MandateConnectorRequest mandateConnectorRequest) {
         return client
                 .target(getDDConnectorUrl(format("/v1/api/accounts/%s/mandates", account.getName())))
                 .request()
                 .accept(MediaType.APPLICATION_JSON)
-                .post(buildAgreementRequestPayload(createAgreementRequest));
+                .post(buildMandateConnectorRequestPayload(mandateConnectorRequest));
     }
 
-    private Response getAgreement(Account account, String agreementId) {
+    Response getMandate(Account account, String mandateExternalId) {
         return client
                 .target(getDDConnectorUrl(format("/v1/api/accounts/%s/mandates/%s",
                         account.getName(),
-                        agreementId)))
+                        mandateExternalId)))
                 .request()
                 .accept(MediaType.APPLICATION_JSON)
                 .get();
     }
 
+    private Entity buildMandateConnectorRequestPayload(MandateConnectorRequest requestPayload) {
+        JsonStringBuilder jsonStringBuilder = new JsonStringBuilder()
+                .add(MandateConnectorRequest.RETURN_URL_FIELD_NAME, requestPayload.getReturnUrl())
+                .add(MandateConnectorRequest.AGREEMENT_TYPE_FIELD_NAME, requestPayload.getAgreementType().toString());
```

Comment:
the `toString()` isn't needed here as the AgreementType enum will automatically serialise to a string

Reply:
since the `toString()` isn't needed, what is the point of `buildMandateConnectorRequestPayload`? we're just passing the MandateConnectorRequest to dd-connector as is.

Reply:
Thanks, removed `toString()` and as discussed we prefer to have the logic in a separate method to be more descriptive